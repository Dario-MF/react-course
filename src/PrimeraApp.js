import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo }) => {
    return (
        <>
            <h1>{saludo}</h1>
            <p>{subtitulo}</p>
        </>
    )
}

PrimeraApp.prototype = {
    saludo: PropTypes.string
}
PrimeraApp.defaultProps = {
    subtitulo: 'Hola soy un subtitulo'
}

export default PrimeraApp
