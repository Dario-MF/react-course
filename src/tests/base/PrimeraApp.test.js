import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import React from 'react'
import PrimeraApp from '../../PrimeraApp'

describe('Tests for component <PrimeraApp /> ', () => {

    /* test('comprobacion de render del subtitulo', () => {
        const saludo = 'Hola soy Dario'
        const { getByText } = render(<PrimeraApp saludo={saludo} />)

        expect(getByText(saludo)).toBeInTheDocument()
    }) */

    test('comprobacion de render del saludo', () => {
        const saludo = 'Hola soy Dario'
        const wrapped = shallow(<PrimeraApp saludo={saludo} />)

        expect(wrapped).toMatchSnapshot()
    })

    test('comprobacion del render del subtitulo', () => {
        const saludo = 'Hola soy Dario'
        const subtitulo = 'Hola soy un subtitulo'

        const wrapped = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subtitulo}
            />)
        const textSaludo = wrapped.find('p').text()
        expect(textSaludo).toBe(subtitulo)

    })



})
