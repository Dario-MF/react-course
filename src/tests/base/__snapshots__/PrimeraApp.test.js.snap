// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for component <PrimeraApp />  comprobacion de render del saludo 1`] = `
<Fragment>
  <h1>
    Hola soy Dario
  </h1>
  <p>
    Hola soy un subtitulo
  </p>
</Fragment>
`;
